internal/runtime/client/client.go

```diff
@@ -432,21 +432,20 @@ func httpCall(ctx context.Context, request, response runtime.Object, opts *httpC
 
 	// use client-go's transport.TLSConfigureFor to ensure good defaults for tls
 	client := http.DefaultClient
-	if opts.config.CABundle != nil {
-		tlsConfig, err := transport.TLSConfigFor(&transport.Config{
-			TLS: transport.TLSConfig{
-				CAData:     opts.config.CABundle,
-				ServerName: extensionURL.Hostname(),
-			},
-		})
-		if err != nil {
-			return errors.Wrap(err, "failed to create tls config")
-		}
-		// this also adds http2
-		client.Transport = utilnet.SetTransportDefaults(&http.Transport{
-			TLSClientConfig: tlsConfig,
-		})
+	tlsConfig, err := transport.TLSConfigFor(&transport.Config{
```

Comment:
If the root ca's allow the call to be trusted it would still be valid (consider a hook hosted using lets encrypt certificates or internal CA which is backed in into the CAPI image)

Reply:
Ah good point. Thx!